### ISSUE TITLE
Improve training form validation

### ISSUE THREAD

**Issue by ic4f** [usegalaxy-eu/tiaas2#7]
Some of the fields lack validation:
- The training request form makes it possible to submit a request for a training that's in the past, as well as a training with an end date that is less than the start date. (also see #6)
- The training identifier should have a unique constraint
- Since the training identifier is used as part of a URL, I think it should consist of alphanumeric characters (no spaces or special chars) EDIT: I see we [clean it](https://github.com/usegalaxy-eu/tiaas2/blob/master/training/views.py#L34); but I wonder whether explicitly validating it and rejecting the form might be less surprising to the user?
- anything else...?

**Comment by hexylena** [comment]
> makes it possible to submit a request for a training that's in the past

In practice I never received a TIaaS request with invalid or incorrect dates. People are pretty careful with these requests and double/triple check them since they have a bit the feeling of a grant application.

> The training identifier should have a unique constraint

Agreed. In practice they always were unique and when they weren't, I manually fixed them with a numeric postfix.

@gmauro processed some trainings which weren't unique though, so when that broke some other things, we couldn't easily retroactively apply that constraint when I realised I missed it. We discussed this, but I don't remember the outcome.

But this field was always intended to be unique, we should enforce it.

> validating it and rejecting the form might be less surprising

sounds good.
